File: 0169-majority-element/0169-majority-element.c
int majorityElement(int* nums, int numsSize) {
    int len=numsSize/2;
    int count[numsSize][2];
    int index=0;
    for(int i=0;i<numsSize;i++){
        int found = 0;
    for (int j = 0; j < index; j++) {
        if (count[j][0] == nums[i]) {
            count[j][1]++;
            found = 1;
            break;   // stop checking once found
        }
    }
    if (!found) {
        count[index][0] = nums[i];
        count[index][1] = 1;
        index++;
    }

    }
    int max_index=0;
    int max_count=0;
    for(int i=0;i<index;i++){
        if(count[i][1]>max_count){
            max_count=count[i][1];
            max_index=i;
        }
    }
    return count[max_index][0];
}

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int num:nums){
            int count=0;
            if(map.containsKey(num)){
                count=map.get(num);
            }
            count++;
            map.put(num,count);
            if(count>nums.length/2)
                return num;
        }
        return -1;
    }
}